core/domain/suggestion_jobs_one_off.py

```diff
@@ -236,3 +237,97 @@ def map(item):
+            # Contributor Dashboard or if the suggestion is not currently in
+            # review.
+            if item.suggestion_type == (
+                    suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT) or (
+                        item.status != suggestion_models.STATUS_IN_REVIEW):
+                return
+            suggestion = suggestion_services.get_suggestion_from_model(item)
+            yield ('suggestion_%s_%s' % (
+                suggestion.suggestion_type, suggestion.language_code), item.id)
+
+        else:
+            if item.can_review_translation_for_language_codes:
+                for language_code in (
+                        item.can_review_translation_for_language_codes):
+                    yield ('reviewer_translation_%s' % language_code, item.id)
+            if item.can_review_questions:
+                yield ('reviewer_question', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        community_contribution_stats = (
+            suggestion_services.get_community_contribution_stats()
+        )
+        keys = key.split('_')
+
+        # Update the reviewer counts.
+        if keys[0] == 'reviewer':
+            if keys[1] == 'question':
+                community_contribution_stats.question_reviewer_count = len(
+                    values)
+            elif keys[1] == 'translation':
+                (
+                    community_contribution_stats
+                    .set_translation_reviewer_count_for_language_code(
+                        keys[-1], len(values))
+                )
+        # Update the suggestion counts.
+        else:
+            # Get the expected key for each suggestion type.
+            question_suggestion_keys = (
+                suggestion_models.SUGGESTION_TYPE_ADD_QUESTION.split('_')
+            )
+            translation_suggestion_keys = (
+                suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT.split('_')
+            )
+            if keys[1] == question_suggestion_keys[0]:
+                community_contribution_stats.question_suggestion_count = len(
+                    values)
+            elif keys[1] == translation_suggestion_keys[0]:
+                (
+                    community_contribution_stats
+                    .set_translation_suggestion_count_for_language_code(
+                        keys[-1], len(values))
+                )
+        # Make sure the counts that have been updated are valid.
+        try:
+            community_contribution_stats.validate()
+        except Exception as e:
+            logging.error(
+                'Community contribution stats failed validation: %s' % e
+            )
+            yield (
+                PopulateCommunityContributionStatsOneOffJob
+                ._VALIDATION_ERROR_KEY,
+                'Community contribution stats failed validation: %s' % e
+            )
+            return
+        suggestion_services.update_community_contribution_stats(
```

Comment:
You should probably have the get + set happen in a transaction, since multiple shards will be running at the same time and you may run into race conditions otherwise.